pkgs/by-name/ec/ec2-instance-selector/package.nix

```
@@ -0,0 +1,39 @@
+{
+  lib,
+  buildGoModule,
+  fetchFromGitHub,
+}:
+
+buildGoModule (finalAttrs: {
+  pname = "ec2-instance-selector";
+  version = "3.1.1";
+
+  src = fetchFromGitHub {
+    owner = "aws";
+    repo = "amazon-ec2-instance-selector";
+    tag = "v${finalAttrs.version}";
+    hash = "sha256-4J66/LiFFeUW20du2clqjz9ozLV+Sn2VVqF9VISXpb0=";
+  };
+
+  vendorHash = "sha256-ocysHrbkmFQ96dEVJvc5YuuBiaXToAcMUUPFiLpMCpU=";
+
+  ldflags = [
+    "-s"
+    "-w"
+    "-X=main.versionID=${finalAttrs.version}"
+    "-X=github.com/aws/amazon-ec2-instance-selector/v3/pkg/selector.versionID=${finalAttrs.version}"
+  ];
+
+  postInstall = ''
+    mv $out/bin/cmd $out/bin/ec2-instance-selector
+  '';
+
+  meta = {
```

will automatically make a pr when it's updated upstream
```suggestion
  passthru.updateScript = nix-update-script { };

  meta = {
```